preview/Win7Msix/Win7MSIXInstaller/ValidateTargetDeviceFamily.cpp

```diff
@@ -0,0 +1,118 @@
+#include <windows.h>
+
+#include <shlobj_core.h>
+#include <CommCtrl.h>
+
+#include "FilePaths.hpp"
+#include "ValidateTargetDeviceFamily.hpp"
+#include "GeneralUtil.hpp"
+#include <TraceLoggingProvider.h>
+#include <VersionHelpers.h>
+
+const PCWSTR ValidateTargetDeviceFamily::HandlerName = L"ValidateRequest";
+
+HRESULT ValidateTargetDeviceFamily::ExecuteForAddRequest()
+{
+    RETURN_IF_FAILED(ParseTargetDeviceFamilyFromPackage());
+    if ((m_targetDeviceFamilyName == L"MSIXCore.Desktop" || m_targetDeviceFamilyName == L"MSIXCore.Server")
+        && IsManifestVersionCompatibleWithOS())
+    {
+        TraceLoggingWrite(g_MsixTraceLoggingProvider,
+            "Validation complete",
+            TraceLoggingLevel(WINEVENT_LEVEL_ERROR));
+    }
+    else
+    {
+        TraceLoggingWrite(g_MsixTraceLoggingProvider,
+            "TDF or OS version did not match",
+            TraceLoggingLevel(WINEVENT_LEVEL_ERROR));

```

Comment:
>WINEVENT_LEVEL_ERROR [](start = 30, length = 20)

as this IS an error, we should return HRESULT_FROM_WIN32(ERROR_INSTALL_REJECTED)

Reply:
Modified to return INSTALL.

Reply:
Updated to return error.